src/backend/joanie/core/api/client/__init__.py

```diff
@@ -528,6 +528,27 @@ def submit_for_signature(self, request, pk=None):  # pylint: disable=no-self-use
 
         return JsonResponse({"invitation_link": invitation_link}, status=HTTPStatus.OK)
 
+    @extend_schema(
+        request=None,
+        responses={
+            204: OpenApiTypes.NONE,
+            422: serializers.ErrorResponseSerializer,
+        },
+    )
+    @action(detail=True, methods=["POST"])
+    def withdraw(self, request, pk=None):  # pylint: disable=no-self-use, invalid-name, unused-argument
+        """Withdraw an order"""
+        order = self.get_object()
+
+        try:
+            order.withdraw()
+        except ValidationError as error:
+            return Response(
```

Comment:
Should we use a JsonResponse instead ? WDYT ?

Reply:
Looking at the response of @lunika , I'd say no https://github.com/openfun/joanie/pull/734#discussion_r1555532833

Reply:
Yes, you are right ! sorry i was confused until this morning about whether choosing the JsonResponse versus the Response. I'd keep the `Response` object.